src/devices/src/virtio/vsock/event_handler.rs

```diff
@@ -22,201 +22,198 @@
+
+        let mut raise_irq = false;
+        if let Err(e) = self.queue_events[TXQ_INDEX].read() {
+            error!("Failed to get vsock tx queue event: {:?}", e);
+        } else {
+            raise_irq |= self.process_tx();
+            // The backend may have queued up responses to the packets we sent during
+            // TX queue processing. If that happened, we need to fetch those responses
+            // and place them into RX buffers.
+            if self.backend.has_pending_rx() {
+                raise_irq |= self.process_rx();
             }
+        }
+        raise_irq
+    }
+
+    fn handle_evq_event(&mut self, event: EpollEvent) -> bool {
+        debug!("vsock: event queue event");
 
-            have_used = true;
-            self.txvq.add_used(&self.mem, head.index, 0);
+        let event_set = event.event_set();
+        if event_set != EventSet::IN {
+            warn!("vsock: evq unexpected event {:?}", event_set);
+            return false;
         }
 
-        have_used
+        if let Err(e) = self.queue_events[EVQ_INDEX].read() {
+            error!("Failed to consume vsock evq event: {:?}", e);
+        }
+        false
+    }
+
+    // Helper function to invoke the backend notify() method from different EventHandler
+    // contexts (handler_read, handle_write, etc). Will not be necessary anymore after
+    // switching to the new polly.
+    fn notify_backend(&mut self, event: EpollEvent) -> bool {
+        debug!("vsock: backend event");
+
+        self.backend.notify(event.event_set());
+        // After the backend has been kicked, it might've freed up some resources, so we
+        // can attempt to send it more data to process.
+        // In particular, if `self.backend.send_pkt()` halted the TX queue processing (by
+        // reurning an error) at some point in the past, now is the time to try walking the
+        // TX queue again.
+        let mut raise_irq = false;
+        raise_irq |= self.process_tx();
+        if self.backend.has_pending_rx() {
+            raise_irq |= self.process_rx();
+        }
+        raise_irq
+    }
+
+    fn handle_activate_event(&self, event_manager: &mut EventManager) {
+        debug!("vsock: activate event");
+        if let Err(e) = self.activate_evt.read() {
+            error!("Failed to consume vsock activate event: {:?}", e);
+        }
+
+        // The subscriber must exist as we previously registered activate_evt via
+        // `interest_list()`.
+        let self_subscriber = event_manager
+            .subscriber(self.activate_evt.as_raw_fd())
+            .unwrap();
+
+        if let Err(e) = event_manager.register(
+            self.queue_events[RXQ_INDEX].as_raw_fd(),
+            EpollEvent::new(
+                EventSet::IN,
+                self.queue_events[RXQ_INDEX].as_raw_fd() as u64,
+            ),
+            self_subscriber.clone(),
+        ) {
+            error!("Failed to register vsock rxq with event manager: {:?}", e);
+        }
```

Comment:
ultra-nit: I think it looks better with unwrap_or_else because the first thing you see is that they're all register operations.
```rust
        event_manager
            .register(
                self.queue_events[RXQ_INDEX].as_raw_fd(),
                EpollEvent::new(
                    EventSet::IN,
                    self.queue_events[RXQ_INDEX].as_raw_fd() as u64,
                ),
                self_subscriber.clone(),
            )
            .unwrap_or_else(|e| {
                error!("Failed to register vsock rxq with event manager: {:?}", e);
            });
```

Reply:
makes sense; done